SetupDataPkg/Library/ConfigVariableListLib/ConfigVariableListLib.c

```diff
@@ -0,0 +1,292 @@
+
+  @retval EFI_INVALID_PARAMETER   Input argument is null.
+  @retval EFI_OUT_OF_RESOURCES    Memory allocation failed.
+  @retval EFI_NOT_FOUND           The requested variable is not found in the active profile.
+  @retval EFI_COMPROMISED_DATA    The profile contains data that does not fit within the structure defined.
+  @retval EFI_SUCCESS             The operation succeeds.
+
+**/
+EFI_STATUS
+EFIAPI
+RetrieveActiveConfigVarList (
+  OUT CONFIG_VAR_LIST_ENTRY  **ConfigVarListPtr,
+  OUT UINTN                  *ConfigVarListCount
+  )
+{
+  return ParseActiveConfigVarList (ConfigVarListPtr, ConfigVarListCount, NULL);
+}
+
+/**
+  Find specified active configuration variable for this platform.
+
+  @param[in]  VarListName       NULL terminated unicode variable name of interest.
+  @param[out] ConfigVarListPtr  Pointer to hold variable list entry from active profile.
+
+  @retval EFI_UNSUPPORTED         Unsupported operation on this platform.
+  @retval EFI_INVALID_PARAMETER   Input argument is null.
+  @retval EFI_OUT_OF_RESOURCES    Memory allocation failed.
+  @retval EFI_NOT_FOUND           The requested variable is not found in the active profile.
+  @retval EFI_COMPROMISED_DATA    The profile contains data that does not fit within the structure defined.
+  @retval EFI_SUCCESS             The operation succeeds.
+
+**/
+EFI_STATUS
+EFIAPI
+QuerySingleActiveConfigUnicodeVarList (
+  IN  CONST CHAR16           *VarName,
+  OUT CONFIG_VAR_LIST_ENTRY  *ConfigVarListPtr
+  )
+{
+  UINTN  ConfigVarListCount = 0;
+
+  if (!VarName || !ConfigVarListPtr) {
+    DEBUG ((DEBUG_ERROR, "%a Null parameter passed\n", __FUNCTION__));
+    return EFI_INVALID_PARAMETER;
+  }
+
+  return ParseActiveConfigVarList (&ConfigVarListPtr, &ConfigVarListCount, VarName);
+}
+
+/**
+  Find specified active configuration variable for this platform.
+
+  @param[in]  VarListName       NULL terminated ascii variable name of interest.
+  @param[out] ConfigVarListPtr  Pointer to hold variable list entry from active profile.
+
+  @retval EFI_UNSUPPORTED         Unsupported operation on this platform.
+  @retval EFI_INVALID_PARAMETER   Input argument is null.
+  @retval EFI_OUT_OF_RESOURCES    Memory allocation failed.
+  @retval EFI_NOT_FOUND           The requested variable is not found in the active profile.
+  @retval EFI_COMPROMISED_DATA    The profile contains data that does not fit within the structure defined.
+  @retval EFI_SUCCESS             The operation succeeds.
+
+**/
+EFI_STATUS
+EFIAPI
+QuerySingleActiveConfigAsciiVarList (
+  IN  CONST CHAR8            *VarName,
+  OUT CONFIG_VAR_LIST_ENTRY  *ConfigVarListPtr
+  )
+{
+  UINTN   ConfigVarListCount = 0;
+  CHAR16  *UniVarName        = NULL;
+
+  if (!VarName || !ConfigVarListPtr) {
+    DEBUG ((DEBUG_ERROR, "%a Null parameter passed\n", __FUNCTION__));
+    return EFI_INVALID_PARAMETER;
+  }
+
+  UniVarName = AllocatePool (AsciiStrLen (VarName) * 2 + 1);

```

Comment:
I thought unicode string has a unicode terminator as well? Also, might want to use a variable for this length to assure consistency.

Reply:
Fixed and changed to AsciiStrSize to include null terminator